### PR TITLE
🔒️ do sanity and staleness check on every oracle call

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/Registry.sol [hunk 1]
```diff
@@ -100,22 +114,83 @@ contract Registry is IRegistry, RegistryGuardian {
...
+
+    /* ///////////////////////////////////////////////////////////////
+                             L2 LOGIC
+    /////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice Checks that the sequencer is not down, and that it is back up for longer than the grace period.
+     * @param creditor The contract address of the Creditor.
+     * @return success Boolean indicating if the sequencer uptime oracle is active.
+     * @return sequencerDown Boolean indicating if the sequencer is down, or is still within the grace period.
```
Comment (j-vp) [inline src/Registry.sol:161]:
```suggestion
     * @return sequencerDown Boolean indicating whether the sequencer is down, or is still within the grace period.
```

File: src/Registry.sol [hunk 2]
```diff
@@ -100,22 +114,83 @@ contract Registry is IRegistry, RegistryGuardian {
...
+        ) {
+            success = true;
+            if (answer == 1 || block.timestamp - startedAt < riskParams[creditor].gracePeriod) {
+                sequencerDown = true;
+            }
+        } catch { }
+    }
+
+    /**
+     * @notice Sets a new sequencer uptime oracle in the case the current oracle is not active.
```
Comment (j-vp) [inline src/Registry.sol:180]: Must differentiate here on 'active oracle - active sequencer' and 'reverting oracle'. In this function, the oracle must revert before a new one can be set. If it's not active, a new one cannot be set.

File: src/Registry.sol [hunk 3]
```diff
@@ -100,22 +114,83 @@ contract Registry is IRegistry, RegistryGuardian {
...
+     * @param sequencerUptimeOracle_ The contract address of the new sequencer uptime oracle.
+     */
+    function setSequencerUptimeOracle(address sequencerUptimeOracle_) external onlyOwner {
+        // Check that the current sequencer uptime oracle is not active.
```
Comment (j-vp) [inline src/Registry.sol:184]:
```suggestion
        // Check that the current sequencer uptime oracle reverts.
```

File: src/Registry.sol [hunk 4]
```diff
@@ -100,22 +114,83 @@ contract Registry is IRegistry, RegistryGuardian {
...
+        (bool success,) = _isSequencerDown(address(0));
+        if (success) revert RegistryErrors.OracleStillActive();
```
Comment (j-vp) [inline src/Registry.sol:186]: Also the 'active' part in the naming must be changed for clarity

File: src/Registry.sol [hunk 5]
```diff
@@ -100,22 +114,83 @@ contract Registry is IRegistry, RegistryGuardian {
...
+
+        // Set the new sequencer uptime oracle.
+        sequencerUptimeOracle = sequencerUptimeOracle_;
+
+        // Check that the new sequencer uptime oracle is active.
```
Comment (j-vp) [inline src/Registry.sol:191]:
```suggestion
        // Check that the new sequencer uptime oracle is not reverting.
```

File: src/Registry.sol [hunk 6]
```diff
@@ -268,6 +343,29 @@ contract Registry is IRegistry, RegistryGuardian {
...
+     * @param minUsdValue_ The minimum USD-value of assets that are taken into account for the Creditor,
+     * denominated in USD with 18 decimals precision.
+     * @param gracePeriod The grace period after the sequencer is back up,
+     * during which no transactions relying on pricing logic can be executed
+     * (increasing open positions, withdrawing collateral and liquidations).
+     * @param maxRecursiveCalls_ The maximum number of calls to different asset modules that are required to process
+     * the deposit/withdrawal/pricing of a single asset.
+     * @dev A minimum USD-value will help to avoid remaining dust amounts in Accounts, which couldn't be liquidated.
+     */
+    function setRiskParameters(address creditor, uint128 minUsdValue_, uint64 gracePeriod, uint64 maxRecursiveCalls_)
```
Comment (j-vp) [inline src/Registry.sol:358]:
```suggestion
    function setRiskParameters(address creditor, uint128 minUsdValue, uint64 gracePeriod, uint64 maxRecursiveCalls)
```

don't think it overwrites it anymore

File: src/Registry.sol [hunk 7]
```diff
@@ -268,6 +343,29 @@ contract Registry is IRegistry, RegistryGuardian {
...
+        external
+        onlyRiskManager(creditor)
+    {
+        riskParams[creditor] = RiskParameters({
+            minUsdValue: minUsdValue_,
```
Comment (j-vp) [inline src/Registry.sol:363]:
```suggestion
            minUsdValue: minUsdValue,
```

File: src/Registry.sol [hunk 8]
```diff
@@ -268,6 +343,29 @@ contract Registry is IRegistry, RegistryGuardian {
...
+            gracePeriod: gracePeriod,
+            maxRecursiveCalls: maxRecursiveCalls_
```
Comment (j-vp) [inline src/Registry.sol:365]:
```suggestion
            maxRecursiveCalls: maxRecursiveCalls
```

**Review by j-vp — APPROVED** [review]
gtg

**Review by zekiblue — APPROVED** [review]
lgtm

**Pull request merged**